docs/proposals/Certificate Revocation Lists.md

```diff
@@ -0,0 +1,213 @@
+# CRL and CRL Cache Design
+
+## Intro
+
+Certificate validation is an essential step during signature validation. Currently Ratify supports checking for revoked certificates through OCSP supported by notation-go library. However, OCSP validation requires internet connection for each validation while CRL could be cached for better performance. As notary-project is adding the CRL support for notation signature validation, Ratify could utilize it.
+
+OCSP URLs can be obtained from the certificate's authority information access (AIA) extension as defined in RFC 6960. If the certificate contains multiple OCSP URLs, then each URL is invoked in sequential order, until a 2xx response is received for any of the URL. For each OCSP URL, wait for a default threshold of 2 seconds to receive an OCSP response. The user may be able to configure this threshold. If OCSP response is not available within the timeout threshold the revocation result will be "revocation unavailable".
+
+If both OCSP URLs and CDP URLs are present, then OCSP is preferred over CRLs. If revocation status cannot be determined using OCSP because of any reason such as unavailability then fallback to using CRLs for revocation check.
+
+## Goals
+
+CRL support, including CRL downloading, validation, and revocation list checks.
+
+- Define a cache provider interface for CRL
```

Comment:
I'm trying to understand what the proposal is for adding file support to the existing cache providers. The provider (currently inmemory/dapr) is entirely unaware of what content it actually is storing. Is the proposal here to implement a 3rd file provider that conforms to the existing cache api? So if a user selects file cache provider then it uses this provider for all cache operations including all manifest descriptor, creds, CRL etc?

Reply:
We are not providing a 3rd provider.
The proposal is more like reuse cache interface for CRL cache user scenario and other cache providers and configurations are not affected.